internal/ethapi/api.go

```diff
@@ -1654,7 +1681,7 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 		"l1Fee":             (*hexutil.Big)(receipt.L1Fee),
 	}
 	// Assign the effective gas price paid
-	if !s.b.ChainConfig().IsLondon(bigblock) {
+	if !s.b.ChainConfig().IsBanach(bigblock) {
```

Comment:
Behavior after London and before Banach:

Original behavior: Enter else branch, fields["effectiveGasPrice"] = `tx.EffectiveGasTipValue(header.BaseFee)` = `tx.GasTipCap()`
New Behavior: Enter if branch, fields["effectiveGasPrice"] = `tx.GasPrice()` = `tx.GasTipCap()`